pkg/kv/kvserver/replica_raft.go

```diff
@@ -3112,9 +3132,19 @@ func (r *Replica) updateLastUpdateTimesUsingStoreLivenessRLocked(
 	for _, desc := range r.descRLocked().Replicas().Descriptors() {
 		// If the replica's store if providing store liveness support, update
 		// lastUpdateTimes to indicate that it is alive.
+		//
+		// Otherwise, mark the replica as unreachable, so that eventually raft is
+		// notified by ReportUnreachable, and the flow transfers to StateProbe. Do
+		// this only the first time, so that addUnreachableRemoteReplica is called
+		// only once per disconnect.
+		//
+		// TODO(pav-kv): there is an overlap between r.mu.lastUpdateTimes and
+		// r.unreachablesMu. Unify them into one mechanism.
 		_, curExp := (*replicaRLockedStoreLiveness)(r).SupportFrom(raftpb.PeerID(desc.ReplicaID))
 		if storeClockTimestamp.ToTimestamp().LessEq(curExp) {
 			r.mu.lastUpdateTimes.update(desc.ReplicaID, r.Clock().PhysicalTime())
+		} else if r.mu.lastUpdateTimes.updateUnreachable(desc.ReplicaID) {
```

Comment:
Wonder if there is a bug here: what if the `storeClockTimestamp` is zero / unset? We're going to unconditionally report replica as unavailable.